Reorganise README for slightly better readability (#24)
mozilla/bigquery-etl: README.md

Recommended practices
===

- Queries

>   - Should name files `sql/table_version.sql` e.g. `sql/clients_daily_v6.sql`

Reviewer: ```suggestion
  - Should be defined in files named as `sql/table_version.sql` e.g. `sql/clients_daily_v6.sql`
```

  - Should not specify a project or dataset in table names to simplify testing
  - Should be (incremental)[#incremental-queries]
  - Should filter input tables on partition and clustering columns
  - Should use `_` prefix in generated column names not meant for output
  - Should not use jinja templating on the query file in Airflow
- UDFs

>   - Should be used for reusability

Reviewer: ```suggestion
  - Should be used to avoid code duplication
```

  - Should use lower snake case names with `udf_` prefix e.g. `udf_mode_last`

>   - Should name files `udfs/function.{sql,js}` e.g. `udfs/udf_mode_last.sql`

Reviewer: ```suggestion
  - Should be defined in files named as `udfs/function.{sql,js}` e.g. `udfs/udf_mode_last.sql`
```

  - Should use `SQL` over `js` for performance

>   - Must not be used for incremental queries with mostly materialized view

Reviewer: ```suggestion
  - Must not be used for incremental queries with a _mostly materialized view_ (defined below)
```

- Requires less airflow configuration
- Will have tooling to automate backfilling
- Will have tooling to replace partitions atomically to prevent duplicate data

> - Will have tooling to generate an optimized mostly materialized view that

Reviewer: ```suggestion
- Will have tooling to generate an optimized _mostly materialized view_ that
```
